### PR TITLE
Perform basic configuration checks

### PR DESCRIPTION
**Pull request by jmelberg-okta**
### Description
- Provide basic configuration validation
  - `issuer`
  - `client_id`
  - `client_secret`
  - `redirect_uri`

#### Resolves
- #277

### CODE REVIEW

File: packages/oidc-middleware/src/ExpressOIDC.js [hunk 1]
```diff
@@ -36,6 +34,7 @@ module.exports = class ExpressOIDC extends EventEmitter {
    * @param {string} [options.scope=openid] The scopes that will determine the claims on the tokens
    * @param {string} [options.response_type=code] The OpenId Connect response type
    * @param {number} [options.maxClockSkew=120] The maximum discrepancy allowed between server clocks in seconds
+   * @param {Object} [options.skipConfigValidation] Skip configuration validation for specific keys
```
Comment (robertjd) [inline packages/oidc-middleware/src/ExpressOIDC.js]: I think we should use `testing.disableHttpsCheck` to be consistent with what we've developed for the management SDKs.

File: packages/oidc-middleware/src/configUtil.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+};
+
+configUtil.assertClientCredentials = (clientIdOrSecret) => {
+  const copyMessage = 'You can copy it from the Okta Developer Console ' +
+    'in the details for the Application you created. ' +
+    `Follow these instructions to find it: ${findAppCredentialsURL}`;
+
+  if (!clientIdOrSecret) {
+    throw new OIDCMiddlewareError('Your client credentials are missing. ' + copyMessage);
```
Comment (robertjd) [inline packages/oidc-middleware/src/configUtil.js]: Why are we conflating these two cases into one error message?

File: packages/oidc-middleware/test/unit/constructor.js [hunk 3]
```diff
@@ -14,15 +14,205 @@ describe('new ExpressOIDC()', () => {
...
+  it('should not throw if https issuer validation is skipped', () => {
+    function createInstance() {
+      new ExpressOIDC({
+        issuer: 'http://foo.com',
+        skipConfigValidation: {
+          https: true
+        }
+      });
+    }
+    const errorMsg = 'Your Okta URL must start with https. Current value: http://foo.com. ' +
```
[the remark this answers is not in the document]
Comment (jmelberg-okta) [reply packages/oidc-middleware/test/unit/constructor.js]: I think its better to assert that the error message matches completely, as we'd want tests to fail if we update the error message in our validation checks.

To clean this up, I've concatenated the message to use a shared `const` with the primary error message.

File: packages/oidc-middleware/src/configUtil.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const findDomainURL = 'https://bit.ly/finding-okta-domain';
+const findAppCredentialsURL = 'https://bit.ly/finding-okta-app-credentials';
+
+configUtil.assertIssuer = (issuer, skip = {}) => {
+  const copyMessage = 'You can copy your domain from the Okta Developer ' +
+    'Console. Follow these instructions to find it: ' + findDomainURL;
+
+  if (!issuer) {
+    throw new OIDCMiddlewareError('Your Okta URL is missing. ' + copyMessage);
+  } else if (!skip.https && !issuer.match(/^https:\/\//g)) {
```
[the remark this answers is not in the document]
Comment (jmelberg-okta) [reply packages/oidc-middleware/src/configUtil.js]: Do we? Disabling `https` is an explicit choice for the developer, as they're aware of the implications of doing so. Adding an error message seems like added noise.

File: packages/oidc-middleware/src/ExpressOIDC.js (hunk 1 above)
Comment (jmelberg-okta) [reply packages/oidc-middleware/src/ExpressOIDC.js]: I originally thought about this, but ultimately decided against it.

In the future, its likely that we will want to skip `clientId` and `clientSecret` validation, as these may not be known when initializing the `ExpressOIDC` object (Dynamic client registration). Nesting the flags under `testing` would not apply to only testing use cases.

`skipConfigValidation` may not be the best name, but I see it as verbose enough to tell users of this library that they're going against the norm by using this object.

File: packages/oidc-middleware/src/configUtil.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (robertjd) [reply packages/oidc-middleware/src/configUtil.js]: We do still want the warning, to be attention grabbing as possible.

File: packages/oidc-middleware/src/ExpressOIDC.js (hunk 1 above)
Comment (robertjd) [reply packages/oidc-middleware/src/ExpressOIDC.js]: We had a long conversation about this, and in the end we decided this should go under the `testing` key to be very clear that this is for a testing case and should be avoided for any kind of real production situation.  Hence the warning I mention above.  We could still add some other key in the future to address the client ID concerns you raise, but we'd still want to break out this https disabling to the `testing` key.

**Pull request merged**